cli/cmd/export.go

```diff
@@ -0,0 +1,128 @@
+/*
+Copyright 2020 Cortex Labs, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cmd
+
+import (
+	"fmt"
+	"os"
+	"path"
+
+	"github.com/cortexlabs/cortex/cli/cluster"
+	"github.com/cortexlabs/cortex/pkg/lib/aws"
+	"github.com/cortexlabs/cortex/pkg/lib/exit"
+	"github.com/cortexlabs/cortex/pkg/lib/telemetry"
+	"github.com/cortexlabs/cortex/pkg/lib/zip"
+	"github.com/cortexlabs/cortex/pkg/types"
+	"github.com/cortexlabs/cortex/pkg/types/spec"
+	"github.com/cortexlabs/cortex/pkg/types/userconfig"
+	"github.com/spf13/cobra"
+)
+
+var _flagExportEnv string
+
+func exportInit() {
+	_exportCmd.Flags().SortFlags = false
+	_exportCmd.Flags().StringVarP(&_flagExportEnv, "env", "e", getDefaultEnv(_generalCommandType), "environment to use")
+}
+
+var _exportCmd = &cobra.Command{
+	Use:   "export API_NAME",
+	Short: "export an API",
+	Args:  cobra.ExactArgs(1),
+	Run: func(cmd *cobra.Command, args []string) {
+		env, err := ReadOrConfigureEnv(_flagExportEnv)
+		if err != nil {
+			telemetry.Event("cli.export")
+			exit.Error(err)
+		}
+		telemetry.Event("cli.export", map[string]interface{}{"provider": env.Provider.String(), "env_name": env.Name})
+
+		err = printEnvIfNotSpecified(_flagExportEnv, cmd)
+		if err != nil {
+			exit.Error(err)
+		}
+
+		if env.Provider == types.LocalProviderType {
+			exit.Error(ErrorNotSupportedInLocalEnvironment())
+		}
+
+		info, err := cluster.Info(MustGetOperatorConfig(env.Name))
+		if err != nil {
+			exit.Error(err)
+		}
+
+		apiRes, err := cluster.GetAPI(MustGetOperatorConfig(env.Name), args[0])
+		if err != nil {
+			exit.Error(err)
+		}
+
+		var apiSpec *spec.API
+
+		if apiRes.RealtimeAPI != nil {
+			apiSpec = &apiRes.RealtimeAPI.Spec
+		} else if apiRes.BatchAPI != nil {
+			apiSpec = &apiRes.BatchAPI.Spec
+		} else if apiRes.TrafficSplitter != nil {
+			apiSpec = &apiRes.TrafficSplitter.Spec
+		}
+
+		baseDir := apiSpec.Name + "-" + apiSpec.ID
+
+		fmt.Println("exporting api to", baseDir)
+
+		err = os.Mkdir(baseDir, os.ModePerm)
+		if err != nil {
+			exit.Error(err)
+		}
+
+		var awsClient *aws.Client
+		if env.AWSAccessKeyID != nil {
+			awsClient, err = aws.NewFromCreds(*info.ClusterConfig.Region, *env.AWSAccessKeyID, *env.AWSSecretAccessKey)
```

Comment:
maybe this will be addressed in the next PR, but I feel like this should follow the cluster order of operations (minus the clusterconfig) since this may need more permissions than the CLI

Another option is to make it a cluster command, e.g. `cortex cluster export --config` and/or `cortex cluster export --name --region` which would export all the APIs (and not take --env), but I'm not sure, we can discuss